File: spec/dummy-bot-python/transformer.py
import json
import sys

while True:
    line = sys.stdin.readline()
    if not line:
        break

    raw_record = json.loads(line)
    transformed_record = {
        'n': raw_record['n'],
        'goodbye': raw_record['hello'].replace('hello', 'goodbye')
    }

    print json.dumps(transformed_record)

File: spec/dummy-broken-bot-ruby/transformer.rb
require 'json'

STDIN.each_line do |line|
  raw_record = JSON.parse(line)

  transformed_record = {
    :n => raw_record['n'],
    :goodbye => raw_record['hello'].sub('hello', 'goodbye')
  }

  puts transformed_record.to_json
end
